vivarium/interface/interactive.py

```diff
@@ -0,0 +1,120 @@
+from math import ceil
+
+from vivarium.framework.configuration import build_simulation_configuration, build_model_specification
+from vivarium.framework.plugins import PluginManager
+from vivarium.framework.engine import SimulationContext
+
+from .utilities import run_from_ipython, log_progress
+
+
+class InteractiveContext(SimulationContext):
+
+    def __init__(self, configuration, components, plugin_manager=None):
+        super().__init__(configuration, components, plugin_manager)
+        self._initial_population = None
+
+    def setup(self):
+        super().setup()
+        self._start_time = self.clock.time
+
+    def initialize_simulants(self):
+        super().initialize_simulants()
+        self._initial_population = self.population.population
+
+    def reset(self):
+        # This is super crude, but should work for a great deal of components.
+        self.population._population = self._initial_population
+        self.clock._time = self._start_time
+
+    def run_for(self, duration, with_logging=True):
+        return self.run_until(self.clock.time + duration, with_logging=with_logging)
+
+    def run_until(self, end_time, with_logging=True):
+        if not isinstance(end_time, type(self.clock.time)):
+            raise ValueError(f"Provided time must be an instance of {type(self.clock.time)}")
+        iterations = int(ceil((end_time - self.clock.time)/self.clock.step_size))
+        self.take_steps(number_of_steps=iterations, with_logging=with_logging)
+        assert self.clock.time - self.clock.step_size < end_time <= self.clock.time
+        return iterations
+
+    def step(self, step_size=None):
```

Comment:
We don't have any guarantees that changing step size on an already configured simulation will work correctly. It probably will but it might be better not to offer the possibility in the first place.

Reply:
Do you mean correctly in terms of implementation or correctly in terms of the results being correct?